src/charm.py

```diff
@@ -47,38 +54,119 @@ def __init__(self, *args):
         self.mongodb = DatabaseRequires(self, relation_name="mongodb", database_name=self.database)
         self.framework.observe(self.mongodb.on.database_created, self._on_database_created)
 
-        self._on_config_changed(None)
+        self.kafka = KafkaRequires(
+            self, relation_name="kafka", topic=self.topic, extra_user_roles=self.extra_user_roles
+        )
+        self.framework.observe(self.kafka.on.topic_created, self._on_topic_created)
 
     def _on_config_changed(self, event):
-        self.database = self.model.config.get("database", "")
-        if self.database == "":
-            self.unit.status = BlockedStatus("The database name is not specified.")
-        elif event:
-            self.unit.status = ActiveStatus(f"database: {self.database}")
+        """Handle on config changed event."""
+        # Read new parameters
+        new_database_name = self.model.config.get("database-name", "")
+        new_topic = self.model.config.get("topic-name", "")
+        new_extra_user_roles = self.model.config.get("extra-user-roles", "")
+
+        # check the
+        if self.database == "" or not self.is_database_related:
```

Comment:
Uhm. I honestly don't like the `if self.database` or `if not self.database`. And relying on implicit behavior. I had (not only me) bad experiences with this in the past.

Reply:
I agree to a point, but there are 2 main reasons not to. 

1. If you accidentally set `self.relation.data[self.app].update({"key": ""})` because we default to empty-strings, it removes the key from the data, which is undesired. 
2. It goes against PEP8 recommendations - https://peps.python.org/pep-0008/#programming-recommendations, specifically:

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false:

and

> Don’t compare boolean values to True or False using ==:

Reply:
Uhm, interesting. I did not know about this PEP nudge. Well, who the hell am I to disagree with PEP and Guido?! Yes, sure. Let's keep things in line with PEP. Thanks for posting this. Learnt something new.